### DISCUSSION TITLE
How to exit all tests if a custom command fails?

### DISCUSSION THREAD

**Discussion by kary-ajrj** [cypress-io/cypress#20662] [Questions and Help] [answered]
- I have created a custom command that calls an API to fetch a fresh/valid token.
- This token is stored in a fixture file to be used by all the specs / tests.  All the specs get the token from fixture file via the before hook.
- The custom command is called by the first spec file in its before hook.
- What I wish to achieve:
   - Do not run any tests if the API (that fetches the token) does not give me a 200 response code.
   OR
   - Do not run any tests if the custom command failed to write a valid JSON response containing the token.

Happy to know if there is a better way to achieve this :)
TIA!

**Comment by jjhelguero** [comment]
There is [`cypress-fail-fast`](https://github.com/javierbrea/cypress-fail-fast) plugin that looks to be what you are looking to solve your question. With this plugin, you can enable the `FAIL_FAST_ENABLED` option to have it globally and have your custom command run in a `beforeEach()` block in your `support/index.js` file so it will be called for each test and should meet your desired achievements.

**Reply by kary-ajrj** [reply]
Thanks @jjhelguero for your response. I am aware of this plugin and have been using it in this repo.
What I wanted to achieve is, exit all tests (across spec files) if the custom command did not execute successfully - either because the API that was called within it did not give a 200 response code or because the custom command was unable to write to a file at the end.

However, as a workaround I am just letting the test fail one by one via the before hook.

**Comment by kary-ajrj** [comment, marked as answer]
https://github.com/cypress-io/cypress/discussions/20662#discussioncomment-2626691
Marking that comment as an answer.
